packages/swapper/src/swappers/ChainflipSwapper/getChainflipTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,276 @@
+import { getEvmTxFees } from "../txFeeHelpers/evmTxFees/getEvmTxFees";
+// import { getUtxoTxFees } from "../txFeeHelpers/utxoTxFees/getUtxoTxFees";
+
+export const getChainflipTradeQuote = async (
+  input: GetTradeQuoteInput,
+  deps: SwapperDeps,
+): Promise<Result<TradeQuote[], SwapErrorRight>> => {
+  const {
+    sellAsset,
+    buyAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmount,
+    affiliateBps: commissionBps,
+  } = input
+
+  if (!isSupportedChainId(sellAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedChainId(buyAsset.chainId)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `unsupported chainId`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { chainId: sellAsset.chainId },
+      }),
+    )
+  }
+
+  if (!isSupportedAsset(sellAsset.chainId, sellAsset.symbol)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${sellAsset.name}' on chainId '${sellAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: sellAsset.chainId, assetId: sellAsset.assetId },
+      }),
+    )
+  }
+
+  if (!isSupportedAsset(buyAsset.chainId, buyAsset.symbol)) {
+    return Err(
+      makeSwapErrorRight({
+        message: `asset '${buyAsset.name}' on chainId '${buyAsset.chainId}' not supported`,
+        code: TradeQuoteError.UnsupportedTradePair,
+        details: { chainId: buyAsset.chainId, assetId: buyAsset.assetId },
+      }),
+    )
+  }
+
+  const sellChainflipChainKey = `${sellAsset.symbol.toLowerCase()}.${chainIdToChainflipNetwork[sellAsset.chainId as KnownChainIds]}`;
+  const buyChainflipChainKey = `${buyAsset.symbol.toLowerCase()}.${chainIdToChainflipNetwork[buyAsset.chainId as KnownChainIds]}`;
+
+  const brokerUrl = deps.config.REACT_APP_CHAINFLIP_API_URL
+  const apiKey = deps.config.REACT_APP_CHAINFLIP_API_KEY;
+  
+  // Subtract the 0.05% BaaS fee to end up at the final displayed commissionBps
+  let serviceCommission = parseInt(commissionBps) - 5;
+  if (serviceCommission < 0)
+    serviceCommission = 0;
+  
+  const maybeQuoteResponse = await chainflipService.get<ChainflipBaasQuoteQuote[]>(
+    `${brokerUrl}/quotes-native?apiKey=${apiKey}&sourceAsset=${sellChainflipChainKey}&destinationAsset=${buyChainflipChainKey}&amount=${sellAmount}&commissionBps=${serviceCommission}`,
+  );
+
+  // TODO: Throw SellAmountBelowMinimum if that's the cause (parse error response)
+  if (maybeQuoteResponse.isErr()) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Quote request failed',
+        code: TradeQuoteError.NoRouteFound,
+      }),
+    )
+  }
+
+  const { data: quoteResponse } = maybeQuoteResponse.unwrap()
+  
+  const getGasFee = async () => {
```

Comment:
As a general notion throughout this PR, we always try to ensure naming reflects what we call "crypto base unit" i.e the smallest unit (e.g wei for ETH i.e 1 ETH would be 1000000000000000000 in base unit) or precision (e.g something like 1.123456789123456789) to avoid human error when dealing with those constants/methods 🙏🏽